src/main/java/org/openstreetmap/atlas/checks/validation/intersections/LineCrossingWaterBodyCheck.java

```diff
@@ -329,29 +368,63 @@ private boolean intersectionsAreExplicit(final GeometricSurface waterbody,
                 .haveExplicitLocationsForIntersections(polygon, lineItem));
     }
 
+    /**
+     * If the waterbody is an area, the intersection is valid. If it's not an area, it's a
+     * multipolygon, and we make sure that either 1. The intersection is at an outer member of the
+     * multipolygon and the outer member was not synthetically created by the atlas slicing process.
+     * It's estimated that if the outer member doesn't have the {@link LastEditTimeTag}, it likely
+     * did not come from an OSM feature, which suggests it was created due to atlas slicing and
+     * would therefore not be a geometrically accurate member of the relation. 2. The intersecting
+     * feature is entirely floating in an outer member of the relation, and not
+     * intersecting/floating in any of the inner members. The inner members are typically islands so
+     * we avoid flagging in the latter scenario. The outer member must still have the LastEditTime
+     * tag
+     *
+     * @param waterbody
+     *            the GeometricSurface representation of the waterbody
+     * @param object
+     *            the AtlasEntity representation of the waterbody
+     * @param intersectingFeature
+     *            the intersecting building/road/line
+     * @return true if either of the 2 above conditions are satisfied, false otherwise
+     */
+    private boolean intersectsValidPartOfWaterbody(final GeometricSurface waterbody,
+            final AtlasObject object, final PolyLine intersectingFeature)
+    {
+        return !(waterbody instanceof MultiPolygon) || ((Relation) object).members().stream()
```

Comment:
🤯

Reply:
I will leave this up to you but this is a bit hard to follow 😄

Reply:
Added some comments and some clarification in the method docstring.. https://github.com/osmlab/atlas-checks/pull/287/commits/ec536cc2f39f151ad87b4e767e0316fa5ba4818f#diff-829f2e37322d264fb3b26e180abc731eR380

Reply:
Awesome! Thanks for the updates